sklearn/discriminant_analysis.py

```diff
@@ -289,6 +290,9 @@ def _solve_lsqr(self, X, y, shrinkage):
            0-471-05669-3.
         """
         self.means_ = _class_means(X, y)
+        if not self.store_covariance:
+            warnings.warn("from version 0.21 'covariance_' will be stored only"
```

Comment:
I think this should only be raised when `covariance_` is accessed and present by error. I.e. do it with a property.